driver/src/test/java/org/neo4j/driver/internal/InternalStatementResultTest.java

```diff
@@ -386,23 +386,89 @@ public void shouldNotPeekIntoTheFutureWhenResultIsEmpty()
         Record future = result.peek();
     }
 
+    @Test
+    public void shouldNotifyResourcesHandlerWhenFetchedViaList()
+    {
+        SessionResourcesHandler resourcesHandler = mock( SessionResourcesHandler.class );
+        StatementResult result = createResult( 10, resourcesHandler );
+
+        List<Record> records = result.list();
+        assertEquals( 10, records.size() );
+
+        verify( resourcesHandler ).onResultConsumed();
+    }
+
+    @Test
+    public void shouldNotifyResourcesHandlerWhenFetchedViaSingle()
+    {
+        SessionResourcesHandler resourcesHandler = mock( SessionResourcesHandler.class );
+        StatementResult result = createResult( 1, resourcesHandler );
+
+        Record record = result.single();
+        assertEquals( "v1-1", record.get( "k1" ).asString() );
+
+        verify( resourcesHandler ).onResultConsumed();
+    }
+
+    @Test
+    public void shouldNotifyResourcesHandlerWhenFetchedViaIterator()
+    {
+        SessionResourcesHandler resourcesHandler = mock( SessionResourcesHandler.class );
+        StatementResult result = createResult( 1, resourcesHandler );
+
+        while ( result.hasNext() )
+        {
+            assertNotNull( result.next() );
+        }
+
+        verify( resourcesHandler ).onResultConsumed();
+    }
+
+    @Test
+    public void shouldNotifyResourcesHandlerWhenSummary()
+    {
+        SessionResourcesHandler resourcesHandler = mock( SessionResourcesHandler.class );
+        StatementResult result = createResult( 10, resourcesHandler );
+
+        assertNotNull( result.summary() );
```

Comment:
maybe summary several times to ensure only called once?
Similar to other methods and also some combination of all the methods? such as result.single(), result.summary(), then result.consume()?